### PR TITLE
pool: improve test coverage for pool client.

### PR DESCRIPTION
**Pull request by dnldd**
This updates stratum errors to provide more detail to clients and adds more tests covering negative paths.

### CODE REVIEW

File: pool/client_test.go [hunk 1]
```diff
@@ -542,78 +938,218 @@ func testClient(t *testing.T, db *bolt.DB) {
...
-	// Fake a bunch of submissions and calculate the hash rate.
-	setMiner(CPU)
-	atomic.StoreInt64(&client.submissions, 50)
-	time.Sleep(time.Second * 2)
-	hash := client.fetchHashRate()
-	if hash == ZeroRat {
-		t.Fatal("expected a non-nil client hash rate")
-	}
+	// Trigger a client timeout by waiting.
+	time.Sleep(time.Second * 3)
```
Comment (jholdstock) [inline pool/client_test.go]: Why is this required?

Comment (dnldd) [reply pool/client_test.go]: Idling here triggers the connection timeout.

File: pool/client.go [hunk 2]
```diff
@@ -167,47 +162,50 @@ func (c *Client) shutdown() {
...
 func (c *Client) claimWeightedShare() error {
+	if c.cfg.SoloPool {
+		return fmt.Errorf("cannot claim shares in solo pool mode")
+	}
 	if c.cfg.ActiveNet.Name == chaincfg.MainNetParams().Name && c.cfg.FetchMiner() == CPU {
-		log.Error("cpu miners are reserved for only simnet testing purposes")
-		return nil
+		return fmt.Errorf("cannot claim shares for cpu miners on mainnet, " +
+			"reserved for simnet testing purposes only")
 	}
```
[the remark this answers is not in the document]
Comment (dnldd) [reply pool/client.go]: The cpu miner can be used for testnet related tests as well, will update the wording.
